server/datastore/mysql/apple_mdm.go

```diff
@@ -2971,6 +2990,11 @@ func generateDesiredStateQuery(entityType string) string {
 	`, func(s string) string { return dynamicNames[s] })
 }
 
+// Number of hours to wait for a user enrollment to exist for a host after its
+// device enrollment. After that duration, the user-scoped profiles will be
+// delivered to the device-channel.
+const hoursToWaitForUserEnrollmentAfterDeviceEnrollment = 2
```

Comment:
Reached out to product for their thoughts on this approach, waiting for Marko to be back to confirm (https://fleetdm.slack.com/archives/C03C41L5YEL/p1750189431001159).